src/server/zset_family.cc

```diff
@@ -1707,6 +1713,7 @@ void ZSetFamily::ZAdd(CmdArgList args, ConnectionContext* cntx) {
   }
   DCHECK(cntx->transaction);
 
+  std::sort(members.begin(), members.end());
```

Comment:
we must sort if we want to benefit from the optimization of adding the next element as last